xbmc/platform/freebsd/CPUInfoFreebsd.cpp

```diff
@@ -245,12 +245,16 @@ float CCPUInfoFreebsd::GetCPUFrequency()
 
 bool CCPUInfoFreebsd::GetTemperature(CTemperature& temperature)
 {
+  if (CCPUInfoPosix::GetTemperature(temperature))
+    return true;
+
```

Comment:
If you are going to override the function, wouldnt you want to do the "native" check before the fallback of CCPUInfoPosix?

I would assume the "native" is in place because its more relevant to the platform?

Reply:
I've explained the change in the OP 🙃

Reply:
I still dont think its the right path. You essentially take away the platform path to provide a potentially niche extension. 

But doesnt affect a platform i use, so im not fussed if you insist on it

Reply:
The problem on Linux is it's a bit of a wack a mole for the cpu temp. There really is no good single interface for that.

Providing a way for the user to override (if the internal method picks the wrong value) or simply adding a way to get their cpu temp (because it's not handled internally) seems like the right behavior to me.

The override here is more for code deduplication purposes than proper inheritance (if there is such a thing).

I hope that helps.

Reply:
Why not split the user supplied temp command into its own method, call that first, do platform logic second, fall back to POSIX finally?

Reply:
> Why not split the user supplied temp command into its own method, call that first, do platform logic second, fall back to POSIX finally?

`CCPUInfoPosix::GetTemperature` is doing the user temp command. So what is there to split out?

Reply:
My bad, sorry for the noise